packages/shared/common/src/internal/fdv2/payloadStreamReader.ts

```diff
@@ -0,0 +1,69 @@
+/* eslint-disable no-underscore-dangle */
+import { EventListener, EventName, LDLogger } from '../../api';
+import { DataSourceErrorKind } from '../../datasource';
+import { ObjProcessors, PayloadListener, PayloadProcessor } from './payloadProcessor';
+
+// Facade interface to contain only ability to add event listeners
```

Comment:
```suggestion
/**
* Interface for an event stream. Only allows listening to events.
*/
```
I think a doc comment is what we want here.

I would avoid facade unless we were proxying. Here we are just defining the contract needed by this component.

Reply:
Done